detox/ios/DetoxXCUITestRunner/DetoxXCUITestRunner/Handlers/ActionHandler.swift

```diff
@@ -7,48 +7,96 @@ import Foundation
 import XCTest
 
 class ActionHandler {
-  func handle(from params: InvocationParams, on element: XCUIElement) throws {
-
-    let exists = element.waitForExistence(timeout: .defaultTimeout)
-    DTXAssert(
-      exists,
-      "Action failed, element with matcher `\(params.matcherDescription)` does not exist"
-    )
+    
+    func findElement(from params: InvocationParams, predicateHandler: PredicateHandler) -> XCUIElement {
+        
+        let element = predicateHandler.findElement(using: params)
 
+      let exists = element.waitForExistence(timeout: .defaultTimeout)
+      DTXAssert(
+        exists,
+        "Action failed, element with matcher `\(params.matcherDescription)` does not exist"
+      )
+        return element;
+    }
+    
+    func getNormalizedCoordinate(from params: InvocationParams) throws -> XCUICoordinate {
+        do {
+            guard let x = Int(params.params?.first ?? "100"), let y = Int(params.params?[1] ?? "100") else {
+                throw Error.missingTypeTextParam
+            }
+            
+            let screenFrame = try XCUIApplication.appUnderTest().frame
+            let normalizedX = CGFloat(x) / screenFrame.width
+            let normalizedY = CGFloat(y) / screenFrame.height
+            let normalizedPoint = CGVector(dx: normalizedX, dy: normalizedY)
+            let coordinate = try XCUIApplication.appUnderTest().coordinate(withNormalizedOffset: normalizedPoint)
```

Comment:
let's avoid repeated calls to `XCUIApplication.appUnderTest()` by storing it in a variable